zk/legacy_executor_verifier/legacy_executor_verifier.go

```diff
@@ -43,25 +56,38 @@ type VerifierResponse struct {
 	Error            error
 }
 
-var ErrNoExecutorAvailable = fmt.Errorf("no executor available")
+type VerifierBundle struct {
+	request  *VerifierRequest
+	response *VerifierResponse
+}
+
+func NewVerifierBundle(request *VerifierRequest, response *VerifierResponse) *VerifierBundle {
+	return &VerifierBundle{
+		request:  request,
+		response: response,
+	}
+}
 
 type ILegacyExecutor interface {
 	Verify(*Payload, *VerifierRequest, common.Hash) (bool, *executor.ProcessBatchResponseV2, error)
 	CheckOnline() bool
 	QueueLength() int
-	CancelAllVerifications()
-	AllowAllVerifications()
+	// CancelAllVerifications()
+	// AllowAllVerifications()
+	AquireAccess()
+	ReleaseAccess()
 }
 
 type WitnessGenerator interface {
 	GenerateWitness(tx kv.Tx, ctx context.Context, startBlock, endBlock uint64, debug, witnessFull bool) ([]byte, error)
 }
 
 type LegacyExecutorVerifier struct {
-	db             kv.RwDB
-	cfg            ethconfig.Zk
-	executors      []ILegacyExecutor
-	executorNumber int
+	db                     kv.RwDB
+	cfg                    ethconfig.Zk
+	executors              []ILegacyExecutor
+	executorNumber         int
+	cancelAllVerifications uint32
```

Comment:
you could make this an `atomic.Bool` to make life a little easier, it gives you Load and Store methods